src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions.cs

```diff
@@ -102,21 +104,54 @@ internal static partial class ICodeDefinitionFactoryExtensions
                 // Special case.  If we're generating a struct constructor, then we'll need
                 // to initialize all fields in the struct, not just the ones we're creating.  To
                 // do that, we call the default constructor.
-                var realFields = containingTypeOpt.GetMembers()
-                                     .OfType<IFieldSymbol>()
-                                     .Where(f => !f.IsStatic);
-                var initializedFields = parameterToExistingFieldMap.Values
-                                            .OfType<IFieldSymbol>()
-                                            .Where(f => !f.IsImplicitlyDeclared && !f.IsStatic);
-                if (initializedFields.Count() < realFields.Count())
+
+                var numFieldsRequiringInitialization = 0;
+
+                var autoProperties = new List<IPropertySymbol>();
+                foreach (var containingTypeField in containingTypeOpt.GetMembers().OfType<IFieldSymbol>())
+                {
+                    if (!containingTypeField.IsStatic)
+                    {
+                        numFieldsRequiringInitialization++;
+
+                        if (containingTypeField.AssociatedSymbol is IPropertySymbol autoProperty)
+                        {
+                            autoProperties.Add(autoProperty);
+                        }
+                    }
+                }
+
+                if (numFieldsRequiringInitialization == 0)
                 {
-                    // We have less field assignments than actual fields.  Generate a call to the
-                    // default constructor as well.
-                    return ImmutableArray<SyntaxNode>.Empty;
+                    return false;
```

Comment:
add comment.  can be simple as "no fields to initialize, so no need to call default constructor"